bin/faucet/src/client.rs

```diff
@@ -45,7 +46,7 @@ pub const DISTRIBUTE_FUNGIBLE_ASSET_SCRIPT: &str =
 /// for the faucet.
 pub struct FaucetClient {
     rpc_api: ApiClient<Channel>,
-    executor: TransactionExecutor<FaucetDataStore, BasicAuthenticator<StdRng>>,
+    executor: TransactionExecutor,
     data_store: FaucetDataStore,
```

Comment:
I think we can make `data_store` here `Arc<FaucetDataStore>` and avoid all the extra cloning discussed in https://github.com/0xPolygonMiden/miden-node/pull/528/files#r1816146982.

This will require changing `FaucetDataStore::update_faucet_account()` to take `self` as immutable, but that shouldn't be a problem since we store `faucet_account` as `RefCell` anyway.

Also, on line 181 below, we don't need to wrap `faucet_account` in `Rc` (just `RefCell` should work fine).